src/layers/learning/fully_connected.cpp

```diff
@@ -42,7 +41,6 @@ void fully_connected_layer<data_layout::DATA_PARALLEL, El::Device::CPU>
   ::setup_matrices(const El::Grid& grid) {
   learning_layer::setup_matrices(grid);
   deallocate_matrices();
-  m_linearity_gradient = new StarMat<El::Device::CPU>(grid);
   m_bias_gradient = new StarMat<El::Device::CPU>(grid);
```

Comment:
Why is the bias gradient still here, and not using a reference like the linearity gradient.

Reply:
See what I wrote in the opening description of this PR.